### ISSUE TITLE
bug in eval generator and predict generator

### ISSUE THREAD

**Issue by nasergh** [keras-team/keras#11830]
Hi
i made a LSTM model with keras(TF)
and train it and save the weight
after that if i load the weight and use 
`scores =model.predict_generator(test_generator,steps=int(test_generator.samples/test_generator.batch_size)) 
`
its give some result
but if i add `score=model.evaluate_generator(test_generator,steps=int(test_generator.samples/test_generator.batch_size))`
before the predict generator the output will change !!!!
if i add another model evaluate generator before the predict generator the output will change again !!

for example : 
with one predict generator output is 0.2 
and with one model evaluation and then use predict generator output is 0.5 
like below:
`score=model.evaluate_generator(test_generator,steps=int(test_generator.samples/test_generator.batch_size))
scores =model.predict_generator(test_generator,steps=int(test_generator.samples/test_generator.batch_size)) `

i don't understand why output changes because we are not in train phase and the weight and state must be constant,
if i run the same code it's give same result but add another eval generator then the output result of predict generator function will change!!!
i am wonder which output is right !? and i can't trust the output now !! 
Thanks

**Comment by gabrieldemarmiesse** [comment]
Can you use `model.get_weights()` and check that all the numpy arrays are equal before and after?

**Comment by nasergh** [comment]
ok i will do 
my question now is LSTM will remember the previous instance?
for example if i give a dog image and classify it and then give the cat image then in hidden state and output state LSTM remember the previous dog image or not?(if i use return sequence = true)
because now i see that LSTM remember something from previous input and use it to make the output of current input.

**Comment by nasergh** [comment]
> Can you use `model.get_weights()` and check that all the numpy arrays are equal before and after?

Hi
i checked
1- Weight is not change!
2- i found that each time you run eval_generator result is same 
but each time you run a predict generator result is changes !!! then problem is in predict generator only and it's not relate to eval generator !
for example first time run predict generator and test accuracy is 0.7435
next time run predict generator accuracy is 0.7420 and ...
why this is happen !????!!!

**Comment by nasergh** [comment]
i recheck with tensorflow.keras 
and i don't have this issue in that !!!!
this is a very serious problem you need to fix it !!
i switch to tensorflow keras from now :)

**Comment by gabrieldemarmiesse** [comment]
We would need a minimal python example to reproduce this. It's impossible to trigger a bug from just a description of the issue. We have many tests running at each commits, it means that this bug is not easy to trigger.

**Comment by nasergh** [comment]
i think your variable are remembering something from previous run
you need to make initial value for the all of your variable !! 🥇

**Comment by Insiyaa** [comment]
I have the same issue. I trained a model of up to 76% validation accuracy on 4 classes. When I use evaluate_generator, the accuracy is 25%. predict_generator gives just one value. When I use .predict() to predict on single examples, I get the correct classifications which means there's some issue with both predict_generator and evaluate_generator...
